components/builder-api/src/server/resources/pkgs.rs

```diff
@@ -913,6 +915,35 @@ pub fn postprocess_extended_package_list(_req: &HttpRequest,
             .body(body)
 }
 
+pub fn postprocess_package_settings(_req: &HttpRequest,
+                                    packages: &[BuilderPackageSetting],
```

Comment:
Not a blocking issue for this PR, but I am pondering if we can potentially use a generic here to de-duplicate the `postprocess_package_settings`, `postprocess_package_list` and `postprocess_extended_package_list` functions which appear to be identical with the exception of the type of packages `Vec` passed in as an argument.

ex.
```
use serde::Serialize;

-pub fn postprocess_package_list(_req: &HttpRequest,
-                                packages: &[BuilderPackageIdent],
-                                count: i64,
-                                pagination: &Query<Pagination>)
-                                -> HttpResponse {
+pub fn postprocess_package_list<T>(_req: &HttpRequest,
+                                   packages: &Vec<T>,
+                                   count: i64,
+                                   pagination: &Query<Pagination>)
+                                   -> HttpResponse
+    where T: Serialize
+{
```

Reply:
It is possible, it's something i was tinkering with and then ultimately scrapped just to get this iteration in. I'll revisit in the a.m. to see if i can get it to work in a sane way.